acme/acme/jws.py

```diff
@@ -12,14 +12,14 @@
 class Header(jose.Header):
     """ACME-specific JOSE Header. Implements nonce, kid, and url.
     """
-    nonce = jose.Field('nonce', omitempty=True, encoder=jose.encode_b64jose)
-    kid = jose.Field('kid', omitempty=True)
-    url = jose.Field('url', omitempty=True)
+    nonce: bytes = jose.field('nonce', omitempty=True, encoder=jose.encode_b64jose)
+    kid: bytes = jose.field('kid', omitempty=True)
+    url: str = jose.field('url', omitempty=True)
 
     # Mypy does not understand the josepy magic happening here, and falsely claims
     # that nonce is redefined. Let's ignore the type check here.
-    @nonce.decoder  # type: ignore
-    def nonce(value: str) -> bytes:  # pylint: disable=no-self-argument,missing-function-docstring
+    @nonce.decoder  # type: ignore[no-redef,attr-defined]
+    def nonce(value: str) -> bytes:  # type: ignore[misc]  # pylint: disable=no-self-argument,missing-function-docstring
```

Comment:
I know you're not changing this, but why is value here of type `str`? Given the encoder and decoder of `nonce`, isn't it always `bytes`?

Reply:
But `jose.decode_b64jose` accepts only `str` right?

Reply:
Ah thanks. I was mixing up `b64encode`/`b64decode` with `encode_b64jose`/`decode_b64jose`.